From 110fec3e41faf47277447f1c56635c5a876c1f96 Mon Sep 17 00:00:00 2001
From: Steffen Maier <maier@linux.ibm.com>
Date: Wed, 26 Jul 2023 12:59:19 +0200
Subject: [PATCH 18/23] feat(ifcfg): minimize s390-specific network
 configuration aspects

Since consolidated s390 network device configuration with zdev is used,
it takes care of all s390-specific settings.
None of the s390-specific ifcfg variables should be used anymore.
NETTYPE and OPTIONS can be removed entirely.
SUBCHANNELS is kept because there are some dependency chains
(probably around the interface identifiying key, which is not HWADDR):

modules.d/35network-legacy/module-setup.sh
parses (sources) each of /etc/sysconfig/network-scripts/ifcfg-*
and if that contains SUBCHANNELS, creates a symlink from the ifcfg file to
/etc/sysconfig/network-scripts/ccw-${SUBCHANNELS}.conf

modules.d/35network-legacy/ifup.sh installed as /sbin/ifup
checks during team setup if a slave interface is an s390 channel-attached
network interface and then parses (sources)
/etc/sysconfig/network-scripts/ccw-${SUBCHANNELS}.conf
from above, if that file exists

Signed-off-by: Steffen Maier <maier@linux.ibm.com>
---
 modules.d/45ifcfg/write-ifcfg.sh | 41 --------------------------------
 1 file changed, 41 deletions(-)

diff --git a/modules.d/45ifcfg/write-ifcfg.sh b/modules.d/45ifcfg/write-ifcfg.sh
index 4ca96e98..a7804b50 100755
--- a/modules.d/45ifcfg/write-ifcfg.sh
+++ b/modules.d/45ifcfg/write-ifcfg.sh
@@ -10,28 +10,10 @@ mkdir -m 0755 -p /tmp/ifcfg/
 # shellcheck disable=SC2174
 mkdir -m 0755 -p /tmp/ifcfg-leases/
 
-get_config_line_by_subchannel() {
-    local CHANNELS
-    local line
-
-    CHANNELS="$1"
-    while read -r line || [ -n "$line" ]; do
-        if strstr "$line" "$CHANNELS"; then
-            echo "$line"
-            return 0
-        fi
-    done < /etc/ccw.conf
-    return 1
-}
-
 print_s390() {
     local _netif
     local SUBCHANNELS
-    local OPTIONS
-    local NETTYPE
-    local CONFIG_LINE
     local i
-    local OLD_IFS
 
     _netif="$1"
     # if we find ccw channel, then use those, instead of
@@ -43,29 +25,6 @@ print_s390() {
     SUBCHANNELS=${SUBCHANNELS%,}
     echo "SUBCHANNELS=\"${SUBCHANNELS}\""
 
-    CONFIG_LINE=$(get_config_line_by_subchannel "$SUBCHANNELS")
-    # shellcheck disable=SC2181
-    [ $? -ne 0 -o -z "$CONFIG_LINE" ] && return 0
-
-    OLD_IFS=$IFS
-    IFS=","
-    # shellcheck disable=SC2086
-    set -- $CONFIG_LINE
-    IFS=$OLD_IFS
-    NETTYPE=$1
-    shift
-    SUBCHANNELS="$1"
-    OPTIONS=""
-    shift
-    while [ $# -gt 0 ]; do
-        case $1 in
-            *=*) OPTIONS="$OPTIONS $1" ;;
-        esac
-        shift
-    done
-    OPTIONS=${OPTIONS## }
-    echo "NETTYPE=\"${NETTYPE}\""
-    echo "OPTIONS=\"${OPTIONS}\""
     return 0
 }
 
-- 
2.42.0

